cdbot/cogs/quoting.py

```diff
@@ -0,0 +1,382 @@
+
+    def multi_quote_dict(self, messages: List[Message], message: Message) -> dict:
+        multi_quote = dict(
+            content_id=messages[0].id,
+            content_link=messages[0].jump_url,
+            channel=messages[0].channel.id,
+            quoted_at=message.created_at,
+            quoted_by=message.author.id,
+            messages=[],
+        )
+        for line in messages:
+            data = dict(original=line.id, content=line.content, author=dict(id=line.author.id, name=str(line.author)))
+            if len(line.embeds) > 0:
+                data["embed"] = line.embeds[0].to_dict()
+            multi_quote["messages"].append(data)
+        return multi_quote
+
+    async def multi_quote_embed(self, multi_quote: dict) -> Embed:
+        embeds = [message.get("embed", None) for message in multi_quote["messages"]]
+        if any(embeds):
+            return list(filter(lambda x: x is not None, embeds))
+        embed = Embed().set_author(name=str(self.bot.user), icon_url=self.bot.user.avatar_url)
+        quoter = await self.get_member_by_id(multi_quote["quoted_by"])
+        channel = self.bot.get_channel(multi_quote["channel"])
+        footer_text = "Quoted by "
+        if quoter:
+            footer_text += str(quoter)
+        else:
+            footer_text += f"ID: {multi_quote['quoted_by']}"
+        footer_text += " in channel "
+        if channel:
+            footer_text += "#" + str(channel)
+        else:
+            footer_text += f"ID: {multi_quote['channel']}"
+        if quoter:
+            embed = embed.set_footer(text=footer_text, icon_url=str(quoter.avatar_url_as(size=32)))
+        else:
+            embed = embed.set_footer(text=footer_text)
+        embed.timestamp = multi_quote["quoted_at"]
+        messages = []
+        for message in multi_quote["messages"]:
+            original = await self.get_member_by_id(message["author"]["id"])
+            messages.append(
+                dict(content=message["content"], user=str(original) if not None else message["author"]["name"])
+            )
+        lines = []
+        for message in messages:
+            line_str = message["user"] + ": "
+            if "```" in message["content"]:
+                line_str += "\n"
+            line_str += message["content"][:1950]
+            lines.append(line_str)
+        embed.description = "```" + "\n".join(lines) + f"```\n[Jump to content]({multi_quote['content_link']})"
+        return embed
+
+    async def generate_single_quote(self, quoted: Message, original: Message):
+        quote = self.quote_dict(quoted, original)
+        embed = await self.quote_embed(quote)
+        return embed, quote
+
+    async def generate_multi_quote(
+        self, channel: TextChannel, from_message: Message, to_message: Message, message: Message
+    ):
+        messages = await self.range_quote(channel, from_message, to_message)
+        if len("".join(message.content for message in messages)) > 2000:
```

Comment:
Is there no better way of handling this case? Can it not be split into multiple consecutive messages, as I'm assuming the 2000 comes from Discord's character limit?

Reply:
At some point in this PR's history, I had included a paginator to make multiple embeds for very long quotes, however, the code was quite messy and so was removed. It can however be added at a later date or now if it's really needed with better code, but I removed purely because the chance of that occuing should be relatively low.

Reply:
Actually, looking at the code, this check can probably be removed, more changed to make it so that it maxes out at 10 quotes rather than having an actual character limit.